platform/viewer/src/lib/DICOMSR/dataExchange.js

```diff
@@ -0,0 +1,33 @@
+import OHIF from '@ohif/core';
+import { retrieveMeasurementFromSR, stowSRFromMeasurements } from './handleSR';
+import { getLatestSRSeries } from './srUtils';
+
+export const retrieveMeasurements = () => {
+  console.info('[DICOMSR] retrieveMeasurements');
+
+  const latestSeries = getLatestSRSeries();
```

Comment:
Getting the latest SR series here causes some series to be skipped while others have their measurements duplicated depending on the random sequence that the series are loaded into the viewer. This happens when there are multiple SR's in the study, potentially from multiple saves of the measurements.

Reply:
When using lazyLoading for the series, we were calling this retrieve measurements several times, because of that, we were loading all SRs. 
I've change the way we call the retrieveMeasrurements method and how we handle the async request of series with lazy load. So now we only call this once we get all series loaded, then we can get the latest one as disired by our team. 

It should be working now.

Reply:
I wonder if it might be worth getting the latest SR that's already loaded from the viewer rather than fetching them again. See the presentation state code here for the componentDidUpdate(). Maybe this case is different than PR or perhaps there is a problem with what I did there.

https://github.com/OHIF/Viewers/pull/1044/files#diff-acf605965e02491d822a993ef89426abR211

Reply:
So, the viewer will load (completely or by lazy loading) all series metaData, here I will need the instance data, which will required this new request.

Reply:
Thanks, so the intent is that the viewer only loads the series metadata, not instance-level information? If so, I'll need to adjust the presentation state PR to match.

Reply:
So, as far as I understood, we do get all study/series metaData and store it inside the `studyMetadataManager`. The instances we only fetch them when we add the series into a viewport, which will have a cache based on imageId and thats a separated thing.